### PR TITLE
[JB][OPS-11614]

### PR DESCRIPTION
**Pull request by barnesjake**
- call /is-valid endpoint
- update tests
- javascript refresh on verifying bank page

### CODE REVIEW

File: app/assets/javascripts/verifying-bank-refresh.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+//refresh the page every 1 seconds (1000ms)
```
Comment (barnesjake) [inline app/assets/javascripts/verifying-bank-refresh.js:1]: Refreshes the page every 1 second, I couldn't find  a way to get this value from config into the js file, so haven't done so for now

File: app/controllers/WeAreVerifyingYourBankAccountController.scala [hunk 2]
```diff
@@ -54,19 +59,34 @@ class WeAreVerifyingYourBankAccountController @Inject() (
...
+      case EventValue.NotValid    => Redirect(routes.RefundRequestNotSubmittedController.get)
+      case EventValue.NotReceived => Ok(views.weAreVerifyingYourBankAccountPage(status, consent_id, bank_reference_id))
+    }
+  }
+
+  //TODO: include more API updates when they're ready
+  private def updateJourneyWithApiCalls(journey: Journey, eventValue: EventValue, bankAccountSummary: BankAccountSummary): Journey = {
+    val hasFinished: HasFinished = eventValue match {
+      case EventValue.Valid       => HasFinished.YesSucceeded
+      case EventValue.NotValid    => HasFinished.YesRefundNotSubmitted
```
Comment (barnesjake) [inline app/controllers/WeAreVerifyingYourBankAccountController.scala:84]: After speaking to Pawel, we spoke about not putting it into a `HasFinished.YesRefundNotSubmitted` state, but after trying to reuse the "Your refund has not been submitted" action, it expects it to be in a final state - it then updates the state within the post action there, if required - so I think this is actually ok to do (set it to `YesRefundNotSubmitted` here)

File: conf/app.routes [hunk 3]
```diff
@@ -7,6 +7,7 @@ GET         /start                                                         contr
 
 GET         /you-cannot-confirm-your-security-details-yet                  controllers.YouCannotConfirmYourSecurityDetailsYetController.get
 
+# techincal error page?
```
Comment (barnesjake) [inline conf/app.routes:10]: This route exists for this page and I got confused by it, so just left a comment that can be removed when we wire up the custom "error" page

File: project/SbtUpdatesSettings.scala [hunk 4]
```diff
@@ -17,7 +17,9 @@ object SbtUpdatesSettings {
     dependencyUpdatesFilter -= moduleFilter("org.julienrf", "play-json-derived-codecs"),
     dependencyUpdatesFilter -= moduleFilter("com.vladsch.flexmark", "flexmark-all"),
     dependencyUpdatesFilter -= moduleFilter("org.scalatestplus.play", "scalatestplus-play"),
-    dependencyUpdatesFilter -= moduleFilter("com.beachape", "enumeratum-play")
+    dependencyUpdatesFilter -= moduleFilter("com.beachape", "enumeratum-play"),
+    //upgrading to version 9.0.0 breaks the javascript on the select your bank page - OPS-11918 to fix this
```
Comment (barnesjake) [inline project/SbtUpdatesSettings.scala:21]: Happy to discuss the reasoning for this with anyone who wants to know about it

File: test/testsupport/stubs/WireMockHelpers.scala [hunk 5]
```diff
@@ -29,170 +29,187 @@ import scala.jdk.CollectionConverters.MapHasAsJava
 
 object WireMockHelpers {
```
Comment (barnesjake) [inline test/testsupport/stubs/WireMockHelpers.scala:31]: Sorry for the changes in this file, but I wanted to wrap/organise the relevant wire mock stubbings into their respective areas, so it's more clear (atleast for me) what kind of endpoint the stubbing is doing (i.e. what verb) without having to look into the method and check

**Review by Lyrain — APPROVED** [review]

File: app/controllers/WeAreVerifyingYourBankAccountController.scala [hunk 6]
```diff
@@ -54,19 +59,34 @@ class WeAreVerifyingYourBankAccountController @Inject() (
 
     for {
       // TODO: Assert status, consent_id & bank_reference_id match that contained within the journey
-      // TODO: Call backend and check what is the outcome of the Ecospend Webhook, if its not there, redirect to itself, if it succeeds and validation ok then progress, if validation fails then redirect to RequestNotSubmitted
-
+      isValid <- p800RefundsExternalApiConnector.isValid(consent_id.fold(missingConsentIdError)(consentId => consentId))
```
Comment (pawelpanasewicz) [inline app/controllers/WeAreVerifyingYourBankAccountController.scala:62]: Better to use `consentId` which is UUID (no errors needed)

Comment (pawelpanasewicz) [reply app/controllers/WeAreVerifyingYourBankAccountController.scala:62]: (also, option.getOrElse(err)) can be faster to achieve the same

File: app/controllers/WeAreVerifyingYourBankAccountController.scala [hunk 7]
```diff
@@ -54,19 +59,34 @@ class WeAreVerifyingYourBankAccountController @Inject() (
...
       // TODO: If API#1133 or API#JF72745 fails, call API#1132 (EPID0771) Case Management Notified
       // TODO: If API#1133 or API#JF72745 fails, redirect to RequestNotSubmitted
-      _ <- journeyService.upsert(journey.update(
-        bankAccountSummary = bankAccountSummary
-      ))
-    } yield Ok(views.weAreVerifyingYourBankAccountPage(status, consent_id, bank_reference_id))
+      newJourney = updateJourneyWithApiCalls(journey, isValid, bankAccountSummary)
+      _ <- journeyService.upsert(newJourney)
+    } yield isValid match {
+      case EventValue.Valid       => Redirect(routes.RequestReceivedController.getBankTransfer)
```
Comment (pawelpanasewicz) [inline app/controllers/WeAreVerifyingYourBankAccountController.scala:74]: Could we rename `EventValue` with `EcospendValidationResult` so it is easier to understand what it does. EventValue is pretty broad in meaning.

File: project/SbtUpdatesSettings.scala (hunk 4 above)
Comment (pawelpanasewicz) [reply project/SbtUpdatesSettings.scala:21]: Yes, let's talk

**Pull request merged**